### ISSUE TITLE
customize loss_batch()

### ISSUE THREAD

**Issue by 674106399** [fastai/fastai#2013]
I'm implementing a model whose head is combined by 3 branches.
So I expect the loss_batch() to be like:

out1, out2, out3, _ = model(*xb)
out = out1 + out2 + 0.1 * out3

loss1 = loss_func(out1, *yb)
loss2 = loss_func(out2, *yb)
loss3 = loss_func(out3, *yb)
loss = loss1 + loss2 + 0.1 * loss3

I modified the source code of loss_batch() in basic_train.py and replaced CallbackHandler().on_loss_begin() with my own on_loss_begin()

May I fork and push my code to your repo?
(I think my code is a little bit ugly...so I wonder whether you can offer a function to customize the loss_batch() )

Best wishes !

**Comment by sgugger** [comment]
`loss_batch` is at the heart of the training loop so we're not going to change it lightly. 
The training loop is however completely customizable with callbacks. During `on_loss_begin`, `last_output` will contain your three outputs and you can save them, return one, then add the two other parts of the loss in `on_backward_begin`. You should look at what `RNNTrainer` does (in callbacks.rnn) to get some example on how to write this.

Please switch on the forum if you need more help with this!
